transport-classes-io_uring/src/main/java/io/netty/channel/uring/AbstractIoUringStreamChannel.java

```diff
@@ -289,6 +291,21 @@ protected int scheduleRead0(boolean first, boolean socketIsEmpty) {
             assert readBuffer == null;
             assert readId == 0;
 
+            final IOUringStreamChannelConfig channelConfig = (IOUringStreamChannelConfig) config();
+            final IoUringIoHandler ioUringIoHandler = (IoUringIoHandler) registration().ioHandler();
+            // Although we checked whether the current kernel supports `register_buffer_ring`
+            // during the initialization of IoUringIoHandler
+            // we still check it again here.
+            // When the kernel does not support this feature, it helps the JIT to delete this branch.
+            // only `first` value is true, we will recv with the buffer ring;
+            if (IoUring.isRegisterBufferRingSupported() && first && channelConfig.isEnableBufferSelectRead()) {
```

Comment:
Why do we this only on `first` @dreamlike-ocean ? I would think we always want to do this.

Reply:
For most loads, when it's not the first read, you only need to call io_uring_enter once to obtain the data. At this point, using the buffer ring seems a bit wasteful. 
 if you use the buffer ring every time you do a recv, I'm afraid there'll be too many ENOBUFF errors, which will actually mess up the performance. 
WDYT？

Reply:
hmm... maybe what we really want to do is basically `scale` this up and down based on how many reads were "successful" per read loop (before `fireChannelReadComplete()`) was called. Basically a bit like `AdaptiveByteBufRecvAllocator` which keeps track of how many bytes we were able to read the last time and based on this change the size of the allocated buffer. 

So for example if you usually are able to do 4 reads before you receive a EWOULDBLOCK you want to schedule all these 4 to use the ring buffer. 

WDYT ?

Reply:
I think what you said is reasonable. I believe we can indeed remove the "first" restriction. 
Perhaps in the future, we can find an adaptive way to handle various situation better.

Reply:
Also thinking a bit more about all of this I wonder if we can remove the POLL_FIRST option and just use polling by default and disable it when a buffer ring is configured. That seems like a sane thing to do and also remove one knob which is nice

Reply:
By default, use polling. 
If  buffer ring is configured, then use  the recv op with IOSQE_BUFFER_SELECT  flag for  first reading. I think it's a good idea.